doc/contributions/tests.md

```diff
@@ -0,0 +1,17 @@
+# Tests
+
+Currently, shadow-utils only provides system tests.
+
+## System tests
+
+These type of tests are written in shell. Unfortunately, the testing framework
+is tightly coupled to the Ubuntu distribution and it can only be run in this
```

Comment:
They were actually written for Debian specifically by the previous maintainers.

It's very possible that I've moved it more across the Ubuntu line, though.

Reply:
Yes, I think you moved them to Ubuntu. I tried to run them in Debian and this is the result:
7 test(s) passed
69 test(s) failed

Reply:
Chuckle.

Thanks for testing.  We really should be able to do better.  "It shouldn't be that hard" :)

Reply:
It shouldn't be, but it is. I have several ideas for improving our current testing situation, but I need some time to gather my thoughts in written form. Probably several issues that I will open up as I go along with the subject.